pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,386 @@
+func (p *K8sPlugin) OnNodeStateAdd(state *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("k8s-plugin OnNodeStateAdd()")
+	return p.OnNodeStateChange(nil, state)
+}
+
+// OnNodeStateChange Invoked when SriovNetworkNodeState CR is updated, return if need dain and/or reboot node
+func (p *K8sPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("k8s-plugin OnNodeStateChange()")
+	needDrain = false
+	needReboot = false
+
+	p.updateTarget.reset()
+	// Check services
+	err = p.isServicesNeedUpdate()
+	if err != nil {
+		return
+	}
+
+	// Check switchdev config
+	var update, remove bool
+	if update, remove, err = utils.WriteSwitchdevConfFile(new); err != nil {
+		glog.Errorf("k8s-plugin OnNodeStateChange():fail to update switchdev.conf file: %v", err)
+		return
+	}
+	if remove {
+		glog.Info("k8s-plugin OnNodeStateChange(): need reboot node to clean switchdev VFs")
+		needDrain = true
+		needReboot = true
+		return
+	}
+	if update {
+		glog.Info("k8s-plugin OnNodeStateChange(): need reboot node to use the up-to-date switchdev.conf")
+		needDrain = true
+		needReboot = true
+		return
+	}
+	if p.updateTarget.needUpdate() {
+		glog.Infof("k8s-plugin OnNodeStateChange(): needDrain to update %q", p.updateTarget)
+		needDrain = true
+	}
+
+	return
+}
+
+// Apply config change
+func (p *K8sPlugin) Apply() error {
+	glog.Info("k8s-plugin Apply()")
+	if err := p.updateSwichdevService(); err != nil {
+		return err
+	}
+
+	for _, systemService := range p.updateTarget.systemServices {
+		if err := p.updateSystemService(systemService); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) readSwitchdevManifestFiles() error {
+	if p.switchdevService == nil {
+		data, err := ioutil.ReadFile(switchdevUnitFile)
+		if err != nil {
+			return err
+		}
+
+		var switchdevServiceFile *ServiceFile
+		if err := yaml.Unmarshal(data, &switchdevServiceFile); err != nil {
+			return err
+		}
+
+		p.switchdevService = &service.Service{
+			Name:    switchdevServiceFile.Name,
+			Path:    "/etc/systemd/system/" + switchdevServiceFile.Name,
+			Content: switchdevServiceFile.Contents,
+		}
+		// Remove run condition form the service
+		conditionOpt := &unit.UnitOption{
+			Section: "Unit",
+			Name:    "ConditionPathExists",
+			Value:   "!/etc/ignition-machine-config-encapsulated.json",
+		}
+		p.switchdevService, err = service.RemoveFromService(p.switchdevService, conditionOpt)
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.switchdevRunScript == nil {
+		data, err := ioutil.ReadFile(configuresSwitchdevScript)
+		if err != nil {
+			return err
+		}
+		if err := yaml.Unmarshal(data, &p.switchdevRunScript); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) readSystemServiceManifestFile(path string) (*service.Service, error) {
```

Comment:
This should move to be in the function ServiceManager file like the compare services

Reply:
This is reading from `bindata/manifests/machine-config` and parsing the yaml file, I don't think this is related to ServiceManager 
WDYT?

Reply:
Why not? because it end with yaml? I think in the future when we will add services if any it will be in the same approach. Don't you think?